### ISSUE TITLE
bindArray doesn't work with repeated binding

### ISSUE THREAD

**Issue by alexanderkjall** [jdbi/jdbi#1686]
The following code doesn't work if the 'on conflict... update' section triggers:

```
        jdbi.registerArrayType(String.class, "text");

        jdbi.useHandle(handle -> {
            Long externalId = handle.createUpdate("INSERT INTO dns_input.record(source, a, aaaa, cname) VALUES (:source, :a, :aaaa, :cname) on conflict(source) do update set a = :a, aaaa = :aaaa, cname = :cname returning id")
                    .bind("source", source)
                    .bindArray("a", String.class, a)
                    .bindArray("aaaa", String.class, aaaa)
                    .bind("cname", cName)
                    .executeAndReturnGeneratedKeys("id")
                    .mapTo(Long.class).one();

        });
```

The query is converted to this before it's sent to postgresql:
```
INSERT INTO dns_input.record(source, a, aaaa, cname) VALUES ($1, $2, $3, $4) on conflict(source) do update set a = $5, aaaa = $6, cname = $7 returning id
```

And argument $2 and $3 contains the correct data, but $5 and $6 does not.

If I modify my code slightly it works:
```
        jdbi.registerArrayType(String.class, "text");

        jdbi.useHandle(handle -> {
            Long externalId = handle.createUpdate("INSERT INTO dns_input.record(source, a, aaaa, cname) VALUES (:source, :a, :aaaa, :cname) on conflict(source) do update set a = :a2, aaaa = :aaaa2, cname = :cname returning id")
                    .bind("source", source)
                    .bindArray("a", String.class, a)
                    .bindArray("aaaa", String.class, aaaa)
                    .bindArray("a2", String.class, a)
                    .bindArray("aaaa2", String.class, aaaa)
                    .bind("cname", cName)
                    .executeAndReturnGeneratedKeys("id")
                    .mapTo(Long.class).one();

        });
```

**Comment by alexanderkjall** [comment]
I wrote two unit tests that displays the problem in the TestSqlArrays class, the first one works and the second one fails:

```

    @Test
    public void testReusedArrayWithInt() throws Exception {
        int[] input = new int[] {2};

        h.inTransaction(t -> {
            t.createUpdate("create table test_reused_array(id int, a int[], unique (id))").execute();

            t.createUpdate("insert into test_reused_array(id, a) values(1, '{1}')").execute();
            t.createUpdate("insert into test_reused_array(id, a) values(1, :array) "
                + "on conflict(id) do update set a=:array")
                .bind("array", input)
                .execute();

            Array result = t.createQuery("select a from test_reused_array where id=1")
                .map((rs, ctx) -> rs.getArray("a")).first();

            t.createUpdate("drop table test_reused_array");

            assertThat(result.getArray()).isEqualTo(input);
            return 1;
        });
    }

    @Test
    public void testReusedArrayWithString() throws Exception {
        List<String> input = Arrays.asList("second");
        h.registerArrayType(String.class, "text");
        h.inTransaction(t -> {
            t.createUpdate("create table test_reused_array(id int, a text[], unique (id))").execute();

            t.createUpdate("insert into test_reused_array(id, a) values(1, '{\"first\"}')").execute();
            t.createUpdate("insert into test_reused_array(id, a) values(1, :array) "
                + "on conflict(id) do update set a=:array")
                .bindArray("array", String.class, input)
                .execute();

            Array result = t.createQuery("select a from test_reused_array where id=1")
                .map((rs, ctx) -> rs.getArray("a")).first();

            t.createUpdate("drop table test_reused_array");

            assertThat(result.getArray()).isEqualTo(input);
            return 1;
        });
    }

```

And some initial debugging seems to indicate that the problem is that two SqlArrayArgument objects are created from the same Iterator, and when the second is created the iterator is already consumed.

I have not yet managed to find the code that creates the iterator.

**Comment by stevenschlansker** [comment]
Thank you for the test case, this makes it a lot easier to hunt down what's going on.  I agree this looks like a bug.

**Comment by stevenschlansker** [comment]
Will be fixed in upcoming `3.14.0`
